src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -331,6 +331,18 @@ internal virtual LexicalSortKey GetLexicalSortKey()
         /// </summary>
         public abstract ImmutableArray<Location> Locations { get; }
 
+        public virtual bool HasLocationInTree(SyntaxTree tree)
```

Comment:
>HasLocationInTree [](http://example.com/codeflow?start=28&length=17)

We already have an API ```IsDefinedInSourceTree```. How is this one different?

Reply:
Primarily that i didn't know that API existed :D.  If that works the same way, i'll remove all of this and just move to that.